### PR TITLE
Refactor fyne

### PR DESCRIPTION
**Pull request by yaowork**
This refactors code on the fyne interface. It improves performance, and simplifies code for adding other pages to the interface. 

It adds the following pages to the main UI. 

- Sync Page 
- Transaction Details Page

It also fixes the crash issue on fyne stated in #324. 

It should be noted that, the fyne library does not currently support scrolling.

Screenshot from 2019-05-02 17-10-18
Screenshot from 2019-05-02 17-10-32
Screenshot from 2019-05-02 17-11-12

### CODE REVIEW

File: fyne/pages/transaction_details.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/decred/dcrd/dcrutil"
+	"github.com/raedahgroup/godcr/app/walletcore"
+	"github.com/raedahgroup/godcr/fyne/widgets"
+)
+
+type TransactionDetails struct {
+	err    error
+	ctx    context.Context
+	wallet walletcore.Wallet
+
```
Comment (ademuanthony) [inline fyne/pages/transaction_details.go]: this empty line can be deleted

**Review by ademuanthony — CHANGES_REQUESTED** [review]
The overview page is randomly rendered automatically from the other pages. Since this PR is refactoring the interface, I think this issue should also be addressed here.

**Comment by yaowork** [comment]
> The overview page is randomly rendered automatically from the other pages. Since this PR is refactoring the interface, I think this issue should also be addressed here.

Other than rendering the Overview page from the sync page (which is intended), i don't experience this

**Comment by itswisdomagain** [comment]
#### Window content does not auto-resize to fill available window space when window is resized.
Look closely enough, you'll see the initial window size having a slightly different background.
Screenshot 2019-05-17 at 8 11 23 AM

#### Re @ademuanthony's comment:
> The overview page is randomly rendered automatically from the other pages. Since this PR is refactoring the interface, I think this issue should also be addressed here.

You can reproduce the issue by starting the app, waiting for sync to complete and the overview page to be loaded, then switch to any other page, turn off your internet for a while (will cause the wallet to lose connection to peers), turn your internet back on (will re-connect to peers, which will re-trigger the sync callbacks and cause the syncCompleted method to be re-called). It's when this method is recalled, that the window "resets" to the overview page. I believe this to also be the cause of #358.

#### Overview page -> Recent activity table -> Hash not properly aligned vertically.
Screenshot 2019-05-17 at 8 18 42 AM

**Pull request closed without merging**